### ISSUE TITLE
Support nullable reference types

### ISSUE THREAD

**Issue by eredmo** [mocklis/mocklis#8]
We should support nullable reference types in Mocklis.

If the Mocklis class itself is declared in a 'nullable disabled' context, no changes are needed.

If the Mocklis class itself is declared in a 'nullable enabled' context, make reference types nullable in the class if they are nullable or oblivious in the interface. (Note that this last point - oblivious declarations. Our goal here differs from how Visual Studio currently generates stubs. If the interface implementation stubs in the code below were generated by visual studio the return value from the Blah method would not be nullable.)

```
#nullable disable
interface ITest
{
    string Blah();
}

#nullable enable
public class Test : ITest
{
    string? Blah()
    {
        throw new NotImplementedException();
    }
}
```

We should also support the 'notnull' type parameter constraint.

**Comment by eredmo** [comment]
All steps for mocked out event handlers should consider the event handlers nullable.

**Comment by eredmo** [comment]
A first version of this is released as part of version 1.1.0-alpha.1. If there are bugs and issues these should be reported as new issues.
